app/models/ems_refresh/save_inventory_helper.rb

```diff
@@ -42,6 +42,7 @@ def save_inventory_multi(association, hashes, deletes, find_key, child_keys = []
       deletes.reload
     end
     deletes = deletes.to_a
+    deletes_index = deletes.index_by { |x| x }
```

Comment:
I guess I could try it, it will be only 1 line of code then

But seems like delete doesn't return the deleted element, but self, so I will need 2 calls when deleting it?

found  = x if deletes.delete?(x) (which calls include? and delete)

Also I would rather run this by the benchmarks, it seems to me like the last changes added like 100s (to 1300s total, for the 70k images). But it might be just a statistical error. I need to start to measure this like @kbrock :-)